File: BookShop.BusinessLogic/Models/DiscountModels/DiscountResponseModel.cs
using System.Collections.Generic;

namespace BookShop.BusinessLogic.Models.DiscountModels
{
    public class DiscountResponseModel
    {
        public int Count { get; set; }
        public List<DiscountModel> DiscountModels { get; set; }

        public DiscountResponseModel()
        {
            DiscountModels = new List<DiscountModel>();
        }
    }
}

File: BookShop.Presentation/wwwroot/ClientApp/src/app/models/discountModels/discountResponseModel.ts
import { DiscountModel } from 'src/app/models/discountModels/discountModel';

export class DiscountResponseModel {
    public count: number;
    public discountModels: DiscountModel[];

    constructor() {
        this.initializeDefaultModel();
    }

    private initializeDefaultModel() {
        this.discountModels = [];
    }
}